### PR TITLE
fix hash compare on verfy to be bytewise

### PR DESCRIPTION
**Pull request by NathanMcCauley**
Signed-off-by: Nathan McCauley <nathan.mccauley@docker.com>

### CODE REVIEW

**Comment by endophage** [comment]
LGTM

File: cmd/notary/tuf.go [hunk 1]
```diff
@@ -257,9 +258,10 @@ func verify(cmd *cobra.Command, args []string) {
 	}
 
 	// Create hasher and hash data
-	stdinHash := fmt.Sprintf("sha256:%x", sha256.Sum256(payload))
-	serverHash := fmt.Sprintf("sha256:%s", target.Hashes["sha256"])
-	if stdinHash != serverHash {
+	stdinHash := sha256.Sum256(payload)
+	serverHash := target.Hashes["sha256"]
```
Comment (aaronlehmann) [inline cmd/notary/tuf.go]: Does this need a check to verify that there's a sha256 entry? The Sprintf would have been resilient to this case.

**Comment by aaronlehmann** [comment]
LGTM, except please add a check that serverHash is not nil

**Comment by aaronlehmann** [comment]
Actually subtle.ConstantTimeCompare won't crash if you pass in nil, so this is fine as is.

**Pull request merged**